web/src/app/escalation-policies/PolicyEditDialog.js

```diff
@@ -24,78 +23,60 @@ const mutation = gql`
   }
 `
 
-export default class PolicyEditDialog extends PureComponent {
-  static propTypes = {
-    escalationPolicyID: p.string.isRequired,
-    onClose: p.func,
+function PolicyEditDialog(props) {
+  const [value, setValue] = useState(null)
+  const { data, editDialogQueryStatus } = useQuery(query, {
+    pollInterval: 0,
+    variables: { id: props.escalationPolicyID },
+  })
+  const defaultValue = {
+    id: props.escalationPolicyID,
+    name: data.escalationPolicy.name,
```

Comment:
This will break if the cache doesn't return a response with the first render call.

The previous code would have rendered a spinner in a random spot while waiting, but we can just render nothing while loading instead:
```js
if (editDialogQueryStatus.loading && !data.escalationPolicy) return null
```

Reply:
Since we can't put that `if` statement above this line (since hooks can't be called conditionally), lets add `?` to these `data` props

e.g.
`data?.escalationPolicy?.name`

We still want the `if (...) return null` below so that we're not filling fields with incorrect info, but if the cache lookup fails, or the data is otherwise invalid, this line and the rest of `defaultValue` will result in a thrown exception otherwise